Source/API/Extras/CBLIncrementalStore.m

```diff
@@ -1229,148 +1629,70 @@ - (CBLQueryEnumerator*) _queryEnumeratorForQuery:(CBLQuery*)query error:(NSError
     return rows;
 }
 
-#pragma mark - Caching
+#pragma mark - Cache
 
-- (void) _setCacheResults:(NSArray*)array forEntity:(NSString*)entityName predicate:(NSPredicate*)predicate
-{
-    NSString *cacheKey = [NSString stringWithFormat:@"%@_%@", entityName, predicate];
-    [_fetchRequestResultCache setObject:array forKey:cacheKey];
+- (NSData*)SHA1:(NSData *)input  {
+    unsigned char digest[SHA_DIGEST_LENGTH];
+    SHA_CTX ctx;
+    SHA1_Init(&ctx);
+    SHA1_Update(&ctx, input.bytes, input.length);
+    SHA1_Final(digest, &ctx);
+    return [NSData dataWithBytes: &digest length: sizeof(digest)];
 }
 
-- (NSArray*) _cachedQueryResultsForEntity:(NSString*)entityName predicate:(NSPredicate*)predicate
-{
-    NSString *cacheKey = [NSString stringWithFormat:@"%@_%@", entityName, predicate];
-    return [_fetchRequestResultCache objectForKey:cacheKey];
+- (NSString *)encodeCacheKey:(id)key {
+    NSData *archive = [NSKeyedArchiver archivedDataWithRootObject:key];
```

Comment:
I found out recently that this approach doesn't work reliably :(
NSKeyedArchiver doesn't always create identical data for equal objects, maybe because dictionaries don't get written in a consistent order. I need to fix this in CBLQueryBuilder too. I'll probably end up converting the NSPredicate to a string instead of archiving it.